src/vos/vos_gc.c

```diff
@@ -808,18 +808,30 @@ gc_reclaim_pool(struct vos_pool *pool, int *credits, bool *empty_ret)
 int
 gc_init_pool(struct umem_instance *umm, struct vos_pool_df *pd)
 {
-	int	i;
+	int		i;
+	umem_off_t	bag_id;
+	int		size;
+	int		rc;
 
 	D_DEBUG(DB_IO, "Init garbage bins for pool="DF_UUID"\n",
 		DP_UUID(pd->pd_id));
 
 	for (i = 0; i < GC_MAX; i++) {
 		struct vos_gc_bin_df *bin = &pd->pd_gc_bins[i];
 
-		bin->bin_bag_first = UMOFF_NULL;
-		bin->bin_bag_last  = UMOFF_NULL;
+		size = offsetof(struct vos_gc_bag_df, bag_items[gc_bag_size]);
+		bag_id = umem_zalloc(umm, size);
+		if (UMOFF_IS_NULL(bag_id))
+			return -DER_NOMEM;
+
+		rc = umem_tx_add_ptr(umm, bin, sizeof(*bin));
+		if (rc != 0)
+			return rc;
+
 		bin->bin_bag_size  = gc_bag_size;
-		bin->bin_bag_nr	   = 0;
+		bin->bin_bag_first = bag_id;
```

Comment:
do we need to modify the processing code to keep this bag around until the pool is destroyed?

Also, do we need something similar for container level bags or is this sufficient?

Reply:
I'm not sure ... perhaps. I'd like to get this change landed to see if it resolves the weekly test failure, but can leave the ticket open to continue the investigation. My understanding is that this issue is not expected to occur in a real production environment because it only happens with really small pools, so haven't dug real deeply into it.